src/app/components/Notification/index.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import { createContext, useContext, useEffect, useState } from 'react'
+import { StatusType, Notification } from 'grommet'
+
+interface NotificationContextProps {
+  readonly notificationVisible: boolean
+  readonly notificationTitle: string
+  readonly notificationStatus: StatusType
+  showNotification(message: string, type?: StatusType): void
+  hideNotification(): void
+}
+
+interface NotificationProviderProps {
+  children: React.ReactNode
+}
+
+const NotificationContext = createContext<NotificationContextProps>({} as NotificationContextProps)
+
+export const NotificationProvider = (props: NotificationProviderProps) => {
+  const [notificationVisible, setNotificationVisible] = useState(false)
+  const [notificationTitle, setNotificationTitle] = useState('')
+  const [notificationStatus, setNotificationStatus] = useState<StatusType>('normal')
+  const showNotification = (title: string, status: StatusType = 'normal') => {
+    setNotificationVisible(true)
+    setNotificationTitle(title)
+    setNotificationStatus(status)
+  }
+  const hideNotification = () => setNotificationVisible(false)
+
+  useEffect(() => {
+    if (notificationVisible) {
+      const timeout = setTimeout(hideNotification, 2000)
+      return () => clearTimeout(timeout)
+    }
+  }, [notificationVisible])
+
+  const context: NotificationContextProps = {
+    notificationVisible,
+    notificationTitle,
+    notificationStatus,
+    showNotification,
+    hideNotification,
+  }
+
+  return (
+    <NotificationContext.Provider value={context}>
+      {props.children}
+      {notificationVisible && (
+        <Notification
+          toast
+          status={notificationStatus}
+          title={notificationTitle}
+          onClose={hideNotification}
+        />
+      )}
+    </NotificationContext.Provider>
+  )
+}
+
+export const useNotification = (): NotificationContextProps => {
```

Comment:
Overall: I don't see enough benefit for abstracting Notification into a context

Reply:
OK I have moved it directly into the `AddressBox`, and removed the context.